pkg/template/servicebroker/provision.go

```diff
@@ -132,17 +165,17 @@ func (b *Broker) ensureBrokerTemplateInstance(namespace, instanceID string, didW
 }
 
 func (b *Broker) Provision(instanceID string, preq *api.ProvisionRequest) *api.Response {
+	if err := b.waitForSync(); err != nil {
+		return api.InternalServerError(err)
+	}
```

Comment:
@csrwng i wonder if this is what we should be doing when we can't find an entry in the cache, instead of retrying the key and/or hitting the api server directly....